test/app-luatest/gh_7000_compat_module_test.lua

```diff
@@ -216,6 +216,26 @@ g.test_help = function()
     t.assert(compat.help())
 end
 
+g.test_is_new_is_old = function()
+    for _, option_def in pairs(definitions) do
+        local name = option_def.name
+        t.assert(compat[name])
+
+        compat[name] = 'new'
```

Comment:
I think we shouldn't change compat options in the main testing code, see #8033.

Reply:
Similar to `test_new_index()`, I switch only dummy options defined in `definitions`, so it is safe.